tt_metal/impl/program/program.cpp

```diff
@@ -131,9 +130,22 @@ class Program_ {
     void allocate_circular_buffers(const Device *device);
 
     bool is_finalized() const;
+    void allocate_kernel_bin_buf_on_device(Device* device);
     void finalize(Device *device);
     bool is_cached() const { return this->cached_; }
+    ProgramBinaryStatus binaries_on_device_status(std::size_t device_id) const {
+        if (this->binaries_on_device_.find(device_id) != this->binaries_on_device_.end()) {
```

Comment:
To avoid an extra lookup:
```
if (auto it = this->binaries_on_device_.find(device_id); it != this->binaries_on_device_.end()) {
  return it->second;
}
```

Reply:
Done